hopper/flash_attn_interface.py

```diff
@@ -174,7 +175,8 @@ def forward(
             causal,
             descale_q=descale_q,
             descale_k=descale_k,
-            descale_v=descale_v,
+            descale_v=descale_v,     
+            gqa_decoding=False,
```

Comment:
I wonder does it make sense to give user an option to enable GQA optimization for general use cases outside of decoding?

e.g. It's generally useful for small seq_len prefill. In this case we don't really need split-kv, but we want to have each threadblock handle multiple Q heads with the same KV head.

Reply:
Furthermore, does it make sense to just enable GQA optimization by default when input is GQA? I feel it won't cause perf regressions even for long sequence length.

Reply:
I feel it might slow things down a bit, but I haven't tried